azure-proto-compute/Extensions/ArmClientExtensions.cs

```diff
@@ -18,12 +18,14 @@ public static class ArmClientExtensions
         /// <param name="resourceId"> The ResourceIdentifier of the resource that is the target of operations. </param>
         /// <returns> Returns an object representing the operations that can be performed over a specific <see cref="AvailabilitySet" />. </returns>
         /// <exception cref="ArgumentException"> ResourceIdentifier provided is not for an AvailabilitySet. </exception>
+        /// <exception cref="ArgumentNullException"> ResourceIdentifier cannot be null. </exception>
         public static AvailabilitySetOperations GetAvailabilitySetOperations(this AzureResourceManagerClient client, ResourceIdentifier resourceId)
         {
+            if (resourceId == null)
+                throw new ArgumentNullException("resourceId cannot be null.");
```

Comment:
pass in the nameof(resourceId) instead of hard coding the name of the parameter

also double check your inputs, this takes in param name and message, make sure you are passing the right thing into the right place